browser/gemini/gemini_service_factory.cc

```diff
@@ -18,7 +17,7 @@ GeminiServiceFactory* GeminiServiceFactory::GetInstance() {
 
 // static
 GeminiService* GeminiServiceFactory::GetForProfile(Profile* profile) {
-  if (brave::IsTorProfile(profile) ||
+  if (profile->IsTor() ||
```

Comment:
Is this check needed any more since IsIncognitoProfile already checks for IsTor internally?

Reply:
yes, it can be removed. I was following the same semantic when removing `brave::IsTorProfile` and there were lots of them to be removed.

Reply:
BTW IsGuestSession should be covered by IsIncognitoProfile too.

Reply:
fixed in 1f3e00e9b8412b5adba921b94a88faedc7b6ff79